syntax = "proto3";

/*
    Prerequisites For python
    - pip install grpcio grpcio-tools

    Generate python code from .proto file
    - python -m grpc_tools.protoc -I. --python_out=./python --grpc_python_out=./python crawler_text_handler.proto

    Prerequisites For go
    - go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
    - go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
    
    Generate go code from .proto file
    - protoc --go_out=. --go-grpc_out=. crawler_text_handler.proto
*/

// Go에서 사용할 패키지 경로 지정
option go_package = "./go";

service CrawlerTextHandler {
  // Bidirectional streaming where Crawler sends URLs and receives summarized data in response
  rpc StreamUrlSummaries (stream UrlRequest) returns (stream SummarizedDataResponse) {}
}

message UrlRequest {
  string url = 1;
}

message SummarizedDataResponse {
  string content = 1; // 기술 블로그는 글자 많아봤자 30KB. string 타입으로 충분
  string summarized_text = 2;
  repeated string hashtags = 3;
}